hummingbot/market/bitfinex/bitfinex_market.pyx

```diff
@@ -0,0 +1,1161 @@
+
+        try:
+            # sell - is negative amount
+            decimal_amount *= -1
+            self.c_start_tracking_order(order_id, trading_pair, order_type, TradeType.SELL, decimal_price, decimal_amount)
+            order_result = await self.place_order(order_id, trading_pair, decimal_amount, False, order_type, decimal_price)
+
+            exchange_order = SubmitOrder.parse(order_result[4][0])
+            exchange_order_id = exchange_order.oid
+
+            tracked_order = self._in_flight_orders.get(order_id)
+
+            if tracked_order is not None:
+                self.logger().info(f"Created {order_type} sell order {order_id} for {decimal_amount} {trading_pair}.")
+                tracked_order.update_exchange_order_id(exchange_order_id)
+
+            self.c_trigger_event(self.MARKET_SELL_ORDER_CREATED_EVENT_TAG,
+                                 SellOrderCreatedEvent(self._current_timestamp,
+                                                       order_type,
+                                                       trading_pair,
+                                                       decimal_amount,
+                                                       decimal_price,
+                                                       order_id))
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            self.c_stop_tracking_order(order_id)
+            order_type_str = "MARKET" if order_type == OrderType.MARKET else "LIMIT"
+            self.logger().network(
+                f"Error submitting sell {order_type_str} order to Bitfinex for "
+                f"{decimal_amount} {trading_pair} {price}.",
+                exc_info=True,
+                app_warning_msg="Failed to submit sell order to Bitfinex. "
+                                "Check API key and network connection."
+            )
+            self.c_trigger_event(self.MARKET_ORDER_FAILURE_EVENT_TAG,
+                                 MarketOrderFailureEvent(self._current_timestamp, order_id, order_type))
+
+    async def execute_cancel(self, trading_pair: str, order_id: str):
+        """
+        Function that makes API request to cancel an active order
+        """
+        try:
+            exchange_order_id = await self._in_flight_orders.get(order_id).get_exchange_order_id()
+            path_url = "auth/w/order/cancel"
+
+            data = {
+                "id": int(exchange_order_id)
+            }
+
+            cancel_result = await self._api_private("post", path_url=path_url, data=data)
+            # return order_id
+            self.logger().info(f"Successfully cancelled order {order_id}.")
+            self.c_stop_tracking_order(order_id)
+            self.c_trigger_event(self.MARKET_ORDER_CANCELLED_EVENT_TAG,
+                                 OrderCancelledEvent(self._current_timestamp, order_id))
+            return order_id
+
+        except IOError as e:
+            if "order not found" in e.message:
```

Comment:
According to BitFinex API documentation, the response for a bad cancel request would simply be an empty JSON.

> https://docs.bitfinex.com/reference#rest-auth-cancel-order